### DISCUSSION TITLE
Question re EvenPolynomial

### DISCUSSION THREAD

**Discussion by dibyendumajumdar** [mjhoptics/ray-optics#70] [General]
https://github.com/mjhoptics/ray-optics/blob/e55a4c199ddc738235813a08c562d18cce310064/src/rayoptics/elem/profiles.py#L594

I am wondering why there are the fields named coef2 ... coef20 when the coefs list is the one that is used.

Also is there a bug in apply_scale_factor() - it doesn't reference the coefs list.

**Comment by mjhoptics** [comment]
As I started adding profile types, I realized that using a list of coefficients, vs using named attributes, was a more generic approach. I never fully removed the named coefs or finished making them interoperate with the coefs list. My current thinking is that the coefs list should be preferred approach going forward.

It appears that if you call apply_scale_factor on the sequential model and follow it with an update_model(), the update_model call will ultimately call update() on the profile which will call gen_coef_list - which will only fill in the coefs if the list hasn't been previously defined. There's a bug! Frankly, this is too tangled up.
